pkg/git/libgit2/checkout.go

```diff
@@ -326,3 +414,36 @@ func buildSignature(s *git2go.Signature) git.Signature {
 		When:  s.When,
 	}
 }
+
+// getBlankRepoAndRemote returns a newly initialized repository, and a remote connected to the provided url.
+// Callers must make sure to call the below defer statements:
+//	defer repo.Free()
+//	defer remote.Free()
+//	defer remote.Disconnect()
```

Comment:
Instead of calling the precise methods outside of our API. I would provide the instruction to the caller to "free the returned objects and close the remote".

Reply:
Added a `free` func to that effect.